eth/tracers/internal/tracetest/util.go

```diff
@@ -62,6 +62,8 @@ func (c *traceContext) toBlockContext(genesis *core.Genesis) vm.BlockContext {
 		Time:        uint64(c.Time),
 		Difficulty:  (*big.Int)(c.Difficulty),
 		GasLimit:    uint64(c.GasLimit),
+		Random:      &genesis.Mixhash,
```

Comment:
There is code below that already sets the random-value:
```
if genesis.Config.TerminalTotalDifficulty != nil && genesis.Config.TerminalTotalDifficulty.Sign() == 0 {
	context.Random = &genesis.Mixhash
}
```
The "random" value nil-ness unfortunately is used by geth in the EVM implementation to determine the isMerge state. This happens in e.g. `runtime.Call`. I think we cannot safely set the random value here, or it would technically be a bug for chain tracing of pre-merge behavior (not the case in op-stack, but creates invalid L1 EVM traces still).

I see `toBlockContext` is only used in tests, but the test-case already has `.config.terminalTotalDifficulty` set to 0, so I am not sure why this change is needed here.

Reply:
This was required to get `isMerge` set for the test when I initially pushed this PR. It is not required anymore, so I rebased and removed this line.